### ISSUE TITLE
Null check throws before generating DataMatrix Image

### ISSUE THREAD

**Issue by brandengunn** [empira/PDFsharp#3]
https://github.com/empira/PDFsharp/blob/e9739749b4fe66e706c02278f4f16dde2ae70fc2/src/foundation/src/PDFsharp/src/PdfSharp/Drawing.BarCodes/CodeDataMatrix.cs#L156

CodeDataMatrix.Render (called internally by XGraphics.DrawMatrixCode) checks for null to decide if it should call DataMatrixImage.GenerateMatrixImage, but the NullabilityHelper throws System.InvalidOperationException as a result of the Null Check.

I don't see a way to generate the image from outside the Render call, as all the related properties and methods are protected.

**Comment by ThomasHoevel** [comment]
The image will be rendered when you use the GDI build.

Which build are you using?

**Comment by brandengunn** [comment]
I was using [PDFsharp-GDI/6.0.0-preview-1](https://www.nuget.org/packages/PDFsharp-GDI/6.0.0-preview-1).

As a further test, I added these lines to the HelloWorld-gdi,PDFsharp app and reproduced the issue (once getting past other compile issues from missing font files.):

```
            // Draw DataMatrixCode
            var codeSize = XUnit.FromMillimeter(10);
            var code = new CodeDataMatrix("0123456789", 12, new XSize(codeSize, codeSize));
            gfx.DrawMatrixCode(code, XBrushes.Black, new XPoint(width / 2 - codeSize / 2, height / 2 - 80));
```

![image](https://user-images.githubusercontent.com/98982252/232818118-41f33220-a04e-4154-89a0-048106a1c878.png)

I then modified [MatrixCode.cs](https://github.com/empira/PDFsharp/blob/c32450e194203d3e03f1f6f777a70470004e39f4/src/foundation/src/PDFsharp/src/PdfSharp/Drawing.BarCodes/MatrixCode.cs#L95-L100) as follows and was able to produce the Hello World pdf:

```
        internal XImage? MatrixImage
        {
            get => _matrixImage;
            set => _matrixImage = value;
        }

        XImage? _matrixImage = null!;
```

![image](https://user-images.githubusercontent.com/98982252/232817138-9f94c480-bdc9-4128-b130-670edc5b42ea.png)

**Comment by brandengunn** [comment]
Sorry, I learned about the issue submission template after writing the above, so here is the MCVE zip that reproduces it in the PdfSharp IssueSubmission-gdi project:

[Issue.zip](https://github.com/empira/PDFsharp/files/11263402/Issue.zip)

**Comment by ThomasHoevel** [comment]
Thanks for the feedback. Fixed it and fix will be included with PDFsharp 6.0.0-preview-2.
